### PR TITLE
Fix test_create_unsubscribe_request_report_creates_batched_report

### PR DESCRIPTION
**Pull request by BlessedDev**
This PR fixes the issues encountered while merging in https://github.com/alphagov/notifications-admin/pull/5157

### CODE REVIEW

File: tests/app/main/views/test_unsubscribe_requests.py [hunk 1]
```diff
@@ -339,8 +341,8 @@ def test_create_unsubscribe_request_report_creates_batched_report(client_request
         SERVICE_ONE_ID,
         {
             "count": 34,
-            "earliest_timestamp": "2024-06-22",
-            "latest_timestamp": "2024-07-01",
+            "earliest_timestamp": ANY,
+            "latest_timestamp": ANY,
```
Comment (quis) [inline tests/app/main/views/test_unsubscribe_requests.py:345]: Using `ANY` here makes the tests pass but I’m not sure if it fixes the problem. The API will still be called with a `datetime` object not a string.

It also makes it easy to introduce bugs in the future, for example I could swap `earliest` and `latest` and the tests would still pass

File: tests/app/main/views/test_unsubscribe_requests.py [hunk 2]
```diff
@@ -320,8 +322,8 @@ def test_create_unsubscribe_request_report_creates_batched_report(client_request
     summary_data = [
         {
             "count": 34,
-            "earliest_timestamp": "2024-06-22",
-            "latest_timestamp": "2024-07-01",
+            "earliest_timestamp": "2024-06-22 10:30:21",
+            "latest_timestamp": "2024-07-01 21:19:21",
```
Comment (quis) [inline tests/app/main/views/test_unsubscribe_requests.py:326]: Can you check the datetime format here is what’s actually returned by the API?

Comment (BlessedDev) [reply tests/app/main/views/test_unsubscribe_requests.py:326]: At this stage the API returns just the `report_id` and I have mocked the API call. The purpose of the test was to ensure that `service_api_client.create_unsubscribe_request_report` was called with the expected values.

Comment (quis) [reply tests/app/main/views/test_unsubscribe_requests.py:326]: If the API is only returning the `id` then that’s all that should be in the mocked data here.

But I think it’s returning more – is it not the data in the `dict` here?
https://github.com/alphagov/notifications-api/blob/4cd15052a18819ecda8aca6734e93365b99c6cc8/app/service/rest.py#L1173-L1187

Comment (BlessedDev) [reply tests/app/main/views/test_unsubscribe_requests.py:326]: Oh I see. This test was specifically for the `create_unsubscribe_request_report` endpoint, there is a separate test that covers the `download_unsubscribe_request_report` endpoint and that test covers what the API returns https://github.com/alphagov/notifications-admin/blob/97e7ab259b5cd38aa984ea257ef26e894b44b069/tests/app/main/views/test_unsubscribe_requests.py#L349.

Comment (quis) [reply tests/app/main/views/test_unsubscribe_requests.py:326]: OK, I can see that is mocked elsewhere. Can you still check the date format here matches what the API returns. I feel like if we don’t get this matching we’ll end up with some BST/UTC/GMT weirdness

Comment (BlessedDev) [reply tests/app/main/views/test_unsubscribe_requests.py]: I have updated both tests test_data to be exact replicas of the what the API returns.

**Review by quis — APPROVED** [review]
Let’s get this merged now to fix the build

**Pull request merged**